lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1948,6 +1974,41 @@ OpFoldResult AtenMulIntOp::fold(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// AtenSubOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenSubOp::fold(ArrayRef<Attribute> operands) {
+  int64_t intLhs, intRhs;
+  if (matchPattern(getOperand(0), m_TorchConstantInt(&intLhs)) && 
+      matchPattern(getOperand(1), m_TorchConstantInt(&intRhs))) {
+    return getI64IntegerAttr(getContext(), intLhs - intRhs);
+  }
+  
+  double floatLhs, floatRhs;
+  if (matchPattern(getOperand(0), m_TorchConstantFloat(&floatLhs)) &&
```

Comment:
can we share this with sub.int/sub.float? E.g. call `AtenSubIntOp::fold` and `AtenSubFloatOp::fold`? Something like:

```
if (!operands[0] || !operands[1])
  return nullptr;
if (operands[0].isa<IntegerAttr>() && operands[1].isa<IntegerAttr>())
  return AtenSubIntOp::fold(operands);
...
```

Reply:
`fold` is not static method of Operation class. We can not invoke this function without constructing a concrete operation.

Reply:
Okay then let's factor out helpers and share the code.

Reply:
It looks like we need to change atenBinaryIntOperatorFoldHelper and also adapt it to use `ArrayRef<Attribute> operands` rather than matchPattern

Reply:
@silvasean  I've reimplemented the `atenBinaryIntOperatorFoldHelper` and added `atenBinaryFloatOperatorFoldHelper` to handle   float cases. In this PR, I don't want to change other codes too much, so I kept the old `atenBinaryIntOperatorFoldHelper` and its usages. If the reimplementation is what you want, I will open another PR to refactor the codes.